### PR TITLE
docs(upgrade): update to new router

### PR DESCRIPTION
**Pull request by Foxandxss**
I updated the tests but we need to be on new world to make those tests work better, such as mock the new router.

### CODE REVIEW

**Comment by wardbell** [comment]
Should we hold until `ngUpgrade` for RC5 arrives?

**Comment by Foxandxss** [comment]
Personally, no.

Apart from this, lots to be done for rc5.  Let's say this is for rc5

**Comment by brandonroberts** [comment]
Router changes LGTM. Some minor issues with spacing between closing brackets.

**Comment by Foxandxss** [comment]
@brandonroberts point them to me so I can fix that.

File: public/docs/_examples/upgrade-phonecat-2-hybrid/ts/app/phone-detail/phone-detail.component.spec.ts [hunk 1]
```diff
@@ -33,19 +30,21 @@ class MockPhone extends Phone {
...
 
-  beforeEachProviders(() => [
-    { provide: Phone, useClass: MockPhone },
-    { provide: '$routeParams', useValue: {phoneId: 'xyz'}},
-    HTTP_PROVIDERS
-  ]);
+  beforeEach(() => {
+      addProviders([
+      { provide: Phone, useClass: MockPhone },
+      { provide: '$routeParams', useValue: {phoneId: 'xyz'}},
```
Comment (brandonroberts) [inline public/docs/_examples/upgrade-phonecat-2-hybrid/ts/app/phone-detail/phone-detail.component.spec.ts]: Closing bracket spacing

File: public/docs/_examples/upgrade-phonecat-3-final/ts/app/app.routes.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// #docregion
+import { provideRouter, RouterConfig } from '@angular/router';
+
+import { PhoneListComponent } from './phone-list/phone-list.component';
+import { PhoneDetailComponent } from './phone-detail/phone-detail.component';
+
+const routes: RouterConfig = [
+  { path: '', redirectTo: 'phones', pathMatch: 'full'},
```
Comment (brandonroberts) [inline public/docs/_examples/upgrade-phonecat-3-final/ts/app/app.routes.ts]: Closing bracket spacing

File: public/docs/_examples/upgrade-phonecat-3-final/ts/app/phone-detail/phone-detail.component.spec.ts [hunk 3]
```diff
@@ -35,24 +32,34 @@ class MockPhone extends Phone {
...
-  beforeEachProviders(() => [
-    { provide: Phone, useClass: MockPhone },
-    { provide: RouteParams, useValue: new RouteParams({phoneId: 'xyz'})},
-    HTTP_PROVIDERS
-  ]);
-  // #enddocregion routeparams
+  beforeEach(() => {
+    addProviders([
+      { provide: Phone, useClass: MockPhone },
+      { provide: ActivatedRoute, useValue: new ActivatedRouteMock({ params: { 'phoneId': 1 } })},
```
Comment (brandonroberts) [inline public/docs/_examples/upgrade-phonecat-3-final/ts/app/phone-detail/phone-detail.component.spec.ts]: Closing bracket spacing

File: public/docs/_examples/upgrade-phonecat-3-final/ts/app/phone-list/phone-list.component.spec.ts [hunk 4]
```diff
@@ -49,21 +32,18 @@ describe('PhoneList', () => {
...
-    RouteRegistry,
-    { provide: Router, useClass: RootRouter },
-    { provide: ROUTER_PRIMARY_COMPONENT, useValue: AppComponent },
-    { provide: Location, useClass: SpyLocation},
-    { provide: Phone, useClass: MockPhone},
-    HTTP_PROVIDERS
-  ]);
+  beforeEach(() => {
+    addProviders([
+      { provide: Location, useClass: SpyLocation},
```
Comment (brandonroberts) [inline public/docs/_examples/upgrade-phonecat-3-final/ts/app/phone-list/phone-list.component.spec.ts]: Closing bracket spacing

File: public/docs/_examples/upgrade-phonecat-2-hybrid/ts/app/core/phone/phone.service.spec.ts [hunk 5]
```diff
@@ -24,29 +22,26 @@ describe('Phone', function() {
...
   }));
 
-  it('should fetch the phones data from `/phones/phones.json`',
-    (done: () => void) => {
+  it('should fetch the phones data from `/phones/phones.json`', (done: () => void) => {
     mockBackend.connections.subscribe((conn: MockConnection) => {
-      conn.mockRespond(new Response(new ResponseOptions({
-        body: JSON.stringify(phonesData)
-      })));
+      conn.mockRespond(new Response(new ResponseOptions({body: JSON.stringify(phonesData)})));
```
Comment (brandonroberts) [inline public/docs/_examples/upgrade-phonecat-2-hybrid/ts/app/core/phone/phone.service.spec.ts]: There isn't any spacing in the object here

**Comment by brandonroberts** [comment]
@Foxandxss I left some comments

**Comment by wardbell** [comment]
This is not in sync with current release. Needs much work. **Do not merge**

**Comment by Foxandxss** [comment]
No longer relevant. New pr is at #2460

**Pull request closed without merging**